### PR TITLE
Add an option for TLS server certs

### PR DESCRIPTION
**Pull request by tgeoghegan**
Allow configuring the trust store used to verify connections to Redis. This is useful when working with something like GCP Memorystore for Redis ([1]), which will issue a self-signed cert for managed Redis instances. With this option, I can use Terraform to create the managed Redis instance, get the server cert that was generated, and render it into the Caddy configuration file.

[1]: https://cloud.google.com/memorystore/docs/redis

### CODE REVIEW

**Comment by pberkel** [comment]
Hi @tgeoghegan thanks for your contribution, this looks like a useful addition to the module.  From what I understand of your code, `tls_server_certs` contains a string representation of a server certificate, is there a specific reason for taking this approach rather than loading the certificate from a file?  It seems that might bloat the configuration file but is potentially more secure.  I also wonder if there is value in supporting both approaches where `tls_server_certs` could contain the string representation of the certificate OR the path to the certificate file on the filesystem?

I'm presuming that you configure Caddy via JSON since your PR contains no provision to unmarshall `tls_server_certs` from a Caddyfile configuration.  In order to provide support for Caddyfile configuration, this PR will need to add a case option to the following switch statement: https://github.com/pberkel/caddy-storage-redis/blob/52960765c5a149bffae9609dac7474702c937a50/caddyfile.go#L80

**Comment by tgeoghegan** [comment]
> From what I understand of your code, `tls_server_certs` contains a string representation of a server certificate

That's correct, it takes one or more certificates in PEM format. I clarified this in a doccomment, with a reference to the underlying `x509.CertPool` API that it uses.

> [I]s there a specific reason for taking this approach rather than loading the certificate from a file? It seems that might bloat the configuration file but is potentially more secure. I also wonder if there is value in supporting both approaches where `tls_server_certs` could contain the string representation of the certificate OR the path to the certificate file on the filesystem?

I did it this way because it's the most natural way to do it in my deployment scenario, where I render a JSON config from Terraform and then mount the resulting file into a container running Caddy, and then rely on Caddy's dynamic config reloading. However, taking a path to a file is also a neat idea, which I've implemented in the commit just tacked on.

For what it's worth, there shouldn't be a security concern in either case: the X.509 certificate only contains a public key and other non-secret values. 

I've also made a change to support the Caddyfile.

Thank you for your review!

File: storage.go [hunk 1]
```diff
@@ -63,22 +65,41 @@ const (
...
+	// https://pkg.go.dev/crypto/tls#Config
+	TlsInsecure bool `json:"tls_insecure"`
+	// TlsServerCertsPEM is a series of PEM encoded certificates that will be
+	// used by the client to validate trust in the Redis server's certificate
+	// instead of the system trust store. May not be specified alongside
+	// `TlsServerCertsPath`. See `x509.CertPool.AppendCertsFromPem` for details.
+	// https://pkg.go.dev/crypto/x509#CertPool.AppendCertsFromPEM
+	TlsServerCertsPEM string `json:"tls_server_certs_pem"`
+	// TlsServerCerts is the path to a file containing a series of PEM encoded
+	// certificates that will be used by the cleint to validate trust in the
```
Comment (pberkel) [inline storage.go:95]:
```suggestion
	// certificates that will be used by the client to validate trust in the
```

File: storage.go [hunk 2]
```diff
@@ -63,22 +65,41 @@ const (
...
+	// by default.
+	// https://pkg.go.dev/crypto/tls#Config
+	TlsInsecure bool `json:"tls_insecure"`
+	// TlsServerCertsPEM is a series of PEM encoded certificates that will be
+	// used by the client to validate trust in the Redis server's certificate
+	// instead of the system trust store. May not be specified alongside
+	// `TlsServerCertsPath`. See `x509.CertPool.AppendCertsFromPem` for details.
+	// https://pkg.go.dev/crypto/x509#CertPool.AppendCertsFromPEM
+	TlsServerCertsPEM string `json:"tls_server_certs_pem"`
+	// TlsServerCerts is the path to a file containing a series of PEM encoded
```
Comment (pberkel) [inline storage.go:94]:
```suggestion
	// TlsServerCertsPath is the path to a file containing a series of PEM encoded
```

Comment (pberkel) [inline storage.go:94]:
```suggestion
	// TlsServerCertsPath is the path to a file containing a series of PEM encoded
```

**Comment by pberkel** [comment]
Changes look good @tgeoghegan just a couple of minor typos, can I also ask you to add a configuration example to the README? Then we should be ready to merge and tag this feature. Thanks!

**Comment by tgeoghegan** [comment]
Sure! I used the heredoc support documented [here](https://caddyserver.com/docs/caddyfile/concepts) to put the long PEM certificate into an example Caddyfile. Thank you for your review and willingness to take this change!

**Comment by pberkel** [comment]
Thanks for your contribution @tgeoghegan the PR has been merged and tagged release [v1.1.0](https://github.com/pberkel/caddy-storage-redis/releases/tag/v1.1.0).

**Pull request merged**